br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -573,6 +593,111 @@ func (m *DupeDetector) RecordIndexConflictError(ctx context.Context, stream DupK
 	return nil
 }
 
+// RetrieveKeyAndValueFromErrFoundDuplicateKeys retrieves the key and value
+// from ErrFoundDuplicateKeys error.
+func RetrieveKeyAndValueFromErrFoundDuplicateKeys(err error) ([]byte, []byte, error) {
+	errString := err.Error()
```

Comment:
instead of using string matching which is less efficient and error-prone, we can use a structure to save the fields. Like https://github.com/pingcap/tidb/blob/332086ebc863b0effb459c1c0eb5af6defced047/pkg/parser/mysql/error.go#L28-L38